### PR TITLE
Allow describe to work on a stopped site

### PR DESCRIPTION
**Pull request by beeradb**
## The Problem:
Currently you cannot run describe on a stopped site. It reports the following error if you try:

`2017/05/17 13:14:27 Could not describe app: ddev site is configured but not currently running`

## The Fix:
No longer check to ensure a site is running, and instead show the describe command with proper "status" regardless of whether it is running or not.

## The Test:
1. Compile the binary from this branch
2. Stop a running site
3. Run `ddev describe`. It's probably a good idea to test both `ddev describe` from the working directory, as well as `ddev describe [sitename]` from outside the working directory.

## Automation Overview:
I've added automation to `local_test.go` to ensure describe reports properly when a site is stopped.

## Related Issue Link(s):
https://github.com/drud/ddev/issues/164

### CODE REVIEW

File: pkg/plugins/platform/utils.go [hunk 1]
```diff
@@ -83,12 +84,18 @@ func RenderAppRow(table *uitable.Table, site App) {
 	if err == nil {
 		appRoot = strings.Replace(appRoot, userDir, "~", 1)
 	}
+	status := site.SiteStatus()
```
Comment (tannerjfco) [inline pkg/plugins/platform/utils.go]: shiny

**Comment by tannerjfco** [comment]
With this change, its possible to describe a site that does not exist:
```
tannerjfco Desktop/wptest1 » ddev rm; ddev describe
Stopping local-wptest1-web ... done
Stopping local-wptest1-dba ... done
Stopping local-wptest1-db ... done
Removing local-wptest1-web ... done
Removing local-wptest1-dba ... done
Removing local-wptest1-db ... done
Network ddev_default is external, skipping
Stopping nginx-proxy ... done
Removing nginx-proxy ... done
Network ddev_default is external, skipping
Successfully removed the wptest1 application.
NAME     TYPE       LOCATION           URL                        STATUS
wptest1  wordpress  ~/Desktop/wptest1  http://wptest1.ddev.local  not found

MySQL Credentials
-----------------
Username:       	root
Password:       	root
Database name:  	data
Connection Info:	wptest1.ddev.local:3306

Other Services
--------------
MailHog:   	http://wptest1.ddev.local:8025
phpMyAdmin:	http://wptest1.ddev.local:8036
```

This isn't necessarily harmful, but worth noting since the intention is to let ddev describe work for stopped sites, not necessarily removed ones. If we want this to be possible, we should probably provide a better status than "not found" and color it red.

**Comment by rickmanelius** [comment]
I concur with @tannerjfco  and thanks for catching that use case. `ddev describe`. Technically the portion that is similar to `ddev list` would still be valid to display because that is information coming directly from a `ddev config`. What is probably not appropriate is showing the mysql/service information because technically there are no containers. The same is true when in a stopped state. Showing the columns and status is valid, but since the services are not accessible, they should not be displayed.

**Comment by beeradb** [comment]
Great catch, @tannerjfco 

Personally, I'm in favor of just printing a message saying the site doesn't exist if it has been RM'd.  This would be consistent with the behavior of list, which does not show removed sites either. 
Another reason to not show removed sites is that we have no way of supporting the `ddev describe [sitename]` format for removed sites, as that requires us pulling the AppRoot from a container label, as currently implemented.

**Comment by tannerjfco** [comment]
@beeradb I'm good with taking the approach you've described. The info isn't very useful for a non-existent site anyways.

**Comment by beeradb** [comment]
@rickmanelius looks like we have a dev quorum (not that @rfay's opinion isn't important!). Does this approach cause any issues for you?

**Comment by rickmanelius** [comment]
Given our current usage of the docker container label to power `ddev list` (instead of a trackable global config), it would make sense to keep `ddev list` and `ddev describe` experiences similar with respect to removed sites. Technically, we could still read a site configuration if in a removed state, but it's probably more valuable to remind the user the site doesn't exist yet in a start/stopped state.

TL;DR. I support the direction by @beeradb and @tannerjfco.

File: pkg/plugins/platform/local_test.go [hunk 2]
```diff
@@ -292,6 +292,27 @@ func TestLocalStop(t *testing.T) {
...
+		cleanup := site.Chdir()
+
+		testcommon.ClearDockerEnv()
+		err := app.Init(site.Dir)
+		assert.NoError(err)
+
+		out, err := app.Describe()
+		assert.NoError(err)
+
+		assert.Contains(out, "stopped", "describe correctly reports stopped on a non-running site.")
```
Comment (tannerjfco) [inline pkg/plugins/platform/local_test.go]: If i read [this](https://github.com/stretchr/testify/blob/master/assert/assertions.go#L621) right, seems like `msgAndArgs` is only output on a fail of the assertion, so we would want the message to further explain the failed assertion.

Comment (beeradb) [inline pkg/plugins/platform/local_test.go]: Completely correct that it is only output on failure. I do always struggle with the voice for these messages. 

I usually phrase it as what I expected to happen since it would be output next to a failure warning.

i.e. `Failed: describe correctly reports stopped on a non-running site` vs `Failed: describe did not correctly report stopped on a non-running site`

I know @rfay has also expressed some uncertainty for these messages. I'm probably OK with either direction, as long as we adopt one and try our best to stick to it.

**Comment by beeradb** [comment]
Updated to respond to the feedback provided. I believe this should be good to go once-again.

**Comment by beeradb** [comment]
Gonna give this 10-15 minutes and then merge if no objections. Speak now or be stuck fixing it later!
